pkg/isodate/date_test.go

```diff
@@ -150,6 +171,153 @@ func TestDivisibleBy(t *testing.T) {
 			expected: true,
 			hasError: false,
 		},
+		{
+			name:     "1 month divisible by 1 day",
+			larger:   "P1M",
+			smaller:  "P1D",
+			expected: true,
+			hasError: false,
+		},
+		{
+			name:     "1 day not divisible by 1 hour (different units)",
+			larger:   "P1D",
+			smaller:  "PT1H",
+			expected: false,
+			hasError: false,
+		},
```

Comment:
Nit: I think this is fine for this PR, as most likely no one will use 1-hour durations (maybe for entitlement periods?).

But 1 hours, 12 hours etc. should be divisible by 1 day.